src/fs.rs

```diff
@@ -65,16 +66,31 @@ where
 
 fn archive_workspace_directory(working_root_dir: &str, execution_id: &str) -> Result<String, std::io::Error> {
     let workspace_dir = crate::fs::root_workspace_directory(working_root_dir, execution_id);
+    let tgz_file_path = format!("{}/.qovery-workspace/{}.tgz", working_root_dir, execution_id);
+    let tgz_file = File::create(tgz_file_path.as_str())?;
 
-    let tar_gz_file_path = format!("{}/.qovery-workspace/{}.tar.gz", working_root_dir, execution_id);
-
-    let tar_gz_file = File::create(tar_gz_file_path.as_str())?;
-
-    let enc = GzEncoder::new(tar_gz_file, Compression::fast());
+    let enc = GzEncoder::new(tgz_file, Compression::fast());
     let mut tar = tar::Builder::new(enc);
-    tar.append_dir_all(execution_id, workspace_dir)?;
 
-    Ok(tar_gz_file_path)
+    let excluded_files: HashSet<&'static str> = vec![
+        ".terraform.lock.hcl",
+        ".terraform",]
+        .into_iter()
+        .collect();
+
+    for entry in WalkDir::new(workspace_dir.clone())
+        .into_iter()
+        .filter_entry(|e| !excluded_files.contains(e.file_name().to_str().unwrap()))
+    {
+        let entry = entry.unwrap();
```

Comment:
Note: I recommend to use expect with a specific message when you can

Reply:
Done 👍